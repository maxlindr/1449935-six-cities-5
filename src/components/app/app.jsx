import React from 'react';
import {Switch, Route, BrowserRouter, Redirect} from 'react-router-dom';
import PropTypes from 'prop-types';
import {offerPropTypes, userPropTypes} from '../../prop-types';
import Main from '../main/main';
import Login from '../login/login';
import FavoritesRouter from '../favorites-router/favorites-router';
import Room from '../room/room';
import {connect} from 'react-redux';

const App = (props) => {
  const {offers, user} = props;

  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/">
          <Main />
        </Route>

        <Route exact path="/login">
          {/* Временно отключено для удобства тестирования */}
          {/* {user ? <Redirect to="/"/> : <Login />} */}
          <Login />
        </Route>

        <Route exact path="/favorites">
          {
            user
              ? (
                <FavoritesRouter
                  offers={offers.filter((offer) => offer.favorite)}
                />
              )
              : <Redirect to="/"/>
          }
        </Route>

        <Route
          exact
          path="/offer/:id"
          render={({match}) => (
            <Room
              offer={offers.find((offer) => offer.id === match.params.id)}
            />
          )}
        />
      </Switch>
    </BrowserRouter>
  );
};

App.propTypes = {
  user: userPropTypes,
  offers: PropTypes.arrayOf(offerPropTypes).isRequired,
};

const mapStateToProps = (state) => ({
  user: state.user,
  offers: state.offers,
});

export {App};
export default connect(mapStateToProps)(App);
