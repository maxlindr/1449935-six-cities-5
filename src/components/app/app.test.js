import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {App} from './app';

const renderer = new ShallowRenderer();

test(`App должен корректно отображаться`, () => {
  renderer.render(
      <App />
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
