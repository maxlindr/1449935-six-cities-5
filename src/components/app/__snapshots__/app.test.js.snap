// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App должен корректно отображаться 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <Memo(Connect(Connect(WithAlertDialog))) />
    </Route>
    <Route
      exact={true}
      path="/login"
    >
      <Memo(Connect(WithRedirectAuthorized)) />
    </Route>
    <Memo(Connect(PrivateRoute))
      exact={true}
      path="/favorites"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/offer/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/error"
    >
      <Memo(Connect(ErrorPage)) />
    </Route>
    <Route>
      <Memo(Connect(ErrorPage))
        message="Resource not found"
      />
    </Route>
  </Switch>
</Router>
`;
